core/src/main/resources/error/error-classes.json

```
@@ -1443,6 +1443,11 @@
           "A correlated outer name reference within a subquery expression body was not found in the enclosing query: <value>"
         ]
       },
+      "CORRELATED_REFERENCE" : {
+        "message" : [
+          "Expressions <sqlExprs> referencing the outer query are not supported outside of WHERE/HAVING clauses."
```

Can we add the sqlExprs at the end of the error message instead of inserting them in the middle?
```suggestion
          "Expressions referencing the outer query are not supported outside of WHERE/HAVING clauses: <sqlExprs>"
```

---

Yes, it really returns a set of expressions separated by `,`

In `CheckAnalysis.scala`, we make the returning `sqlExprs`:

```scala
messageParameters = Map("sqlExprs" -> exprs.map(toSQLExpr).mkString(",")))
```

---

> Can we add the sqlExprs at the end of the error message instead of inserting them in the middle?

Sounds good. Let me address the comment!